### PR TITLE
Fixes for CI downloads

### PR DESCRIPTION
**Pull request by lebeg**
## Description ##

Fixed for CI verification builds.

### Changes ###
- Checked that silent curl downloads succeeded
- Added curl download retries
- Replaced model download links from dmlc to mxnet

### CODE REVIEW

**Comment by lebeg** [comment]
Yes, you can find it [here](https://github.com/apache/incubator-mxnet/pull/14504/files#diff-6147d49d6089c9bb27fd3b6bd7c1e1a3R26). There seem to be other issues still, will need to take a look into those as well.

**Comment by lanking520** [comment]
@lebeg The static build failure was caused by no default CURL option specified. Without `-s -L`, it won't work properly.

**Comment by lebeg** [comment]
Sorry, I didn’t understand - you are saying that it wouldn’t work without ‘-s’?
The ‘-L’ option is specified in the template by ‘—local’

**Comment by lanking520** [comment]
> Sorry, I didn’t understand - you are saying that it wouldn’t work without `-s `?
> The `-L` option is specified in the template by `—local`.

Might be -s. Please put the environment variable into `make_shared_dependeices.sh`

**Comment by pinaraws** [comment]
@mxnet-label-bot add[pr-awaiting-response]

**Comment by pinaraws** [comment]
@mxnet-label-bot add[CI]

**Comment by lebeg** [comment]
> Might be -s. Please put the environment variable into `make_shared_dependeices.sh`

`-s` is for 'silent' and is the exact reason why CI is not propagating the download error correctly.

**Comment by lebeg** [comment]
> The `-L` option is specified in the template by `—local`.

@lanking520 Sorry, I think I caused some confusion, I meant the `--location` option. I brought the `--silent` option back, but made a check for the downloaded.

**Comment by lebeg** [comment]
> These still have references to that data.dmlc.ml broken link
> 
> `tools/coreml/README.md`
> 
> `example/rcnn/README.md`
> 
> `docs/model_zoo/index.md`
> 
> `tools/coreml/test/test_mxnet_models.py`
> 
> `R-package/tests/testthat/get_data.R`
> 
> `julia/models/Inception/get.sh`
> 
> `cpp-package/example/feature_extract/run.sh`

I could just change `data.dmlc.ml` to `data.mxnet.io` in the README files, but I have checked that some folders mentioned don't exist. For example `http://data.dmlc.ml/models/imagenet/nin/` in `tools/coreml/README.md`.

I'm not sure I want to make those changes in this PR.

File: tools/dependencies/make_shared_dependencies.sh [hunk 1]
```diff
@@ -20,11 +20,37 @@
 # This is a convenience script for calling the build scripts of all dependency libraries.
 # Environment variables should be set beforehand.
 
+set -ex
+
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
 
+download () {
```
Comment (lanking520) [inline tools/dependencies/make_shared_dependencies.sh]: Awesome config! 👍

File: cpp-package/example/get_data.sh [hunk 2]
```diff
@@ -14,29 +16,48 @@
...
 
-if [ ! -d "./data" ]; then
-    mkdir data
-fi
+set -e
+
+mkdir -p data/mnist_data
+cd data/mnist_data
+
+download () {
```
Comment (lanking520) [inline cpp-package/example/get_data.sh]: Seemed not properly configured the file name.

Comment (lebeg) [reply cpp-package/example/get_data.sh]: The error was from incorrect `-o` supplied to the download function.

**Comment by gavinmbell** [comment]
There are two download() functions here.  Can we put that download function in a library type of script and then use it accordingly - to adhere to a DRY (Don't Repeat Yourself) sensibility?

The diff seems to be with unzipping activities.  I realize one is in 'examples' (cpp-package/example/get_data.sh) and the other under tools (tools/dependencies/make_shared_dependencies.sh) - perhaps these are disparate enough things to warrant a copy and paste - I don't know.  Just flagging this.

**Comment by lebeg** [comment]
Created ports for release branches:

* [#14525](https://github.com/apache/incubator-mxnet/pull/14525) Fixes for CI downloads (v1.3.x)
* [#14526](https://github.com/apache/incubator-mxnet/pull/14526) Fixes for CI downloads (v1.4.x)

**Comment by kwanking** [comment]
> ## Description
> 
> Fixed for CI verification builds.
> ### Changes
> 
>     * Checked that silent curl downloads succeeded
> 
>     * Added curl download retries
> 
>     * Replaced model download links from dmlc to mxnet



> ## Description
> 
> Fixed for CI verification builds.
> ### Changes
> 
>     * Checked that silent curl downloads succeeded
> 
>     * Added curl download retries
> 
>     * Replaced model download links from dmlc to mxnet



> ## Description
> 
> Fixed for CI verification builds.
> ### Changes
> 
>     * Checked that silent curl downloads succeeded
> 
>     * Added curl download retries
> 
>     * Replaced model download links from dmlc to mxnet

when run ci/build.py, there is a error,  how do I let timeout be long?  thanks!

build.py: 2019-04-24 13:38:36,494Z WARNING Exception: Command '['docker', 'pull', 'mxnetci/build.android_armv7']' timed out after 900 seconds, Retrying in 50 seconds..

**Pull request merged**